components/feature/session/src/main/java/mozilla/components/feature/session/SessionStorage.kt

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session
+
+import mozilla.components.browser.session.Session
+import mozilla.components.concept.engine.Engine
+
+/**
+ * Storage component for browser and engine sessions.
+ */
+interface SessionStorage {
+
+    /**
+     * Persists the current storage state (all active sessions).
+     *
+     * @param selectedSession the currently selected session, optional.
+     * @return true if the state was persisted, otherwise false.
+     */
+    fun persist(selectedSession: Session? = null): Boolean
+
+    /**
+     * Restores the storage state by reading from the latest persisted version.
+     *
+     * @param engine the engine instance to use. Required to create new engine
+     * sessions.
+     * @return list of all restored sessions
+     */
+    fun restore(engine: Engine): List<SessionProxy>
```

Comment:
Just a thought: I was wondering whether this should just restore some "Snapshot" object that contains everything (like the serialized data) instead of having multiple getters.